docs/plugins/npe2_manifest_specification.md

```diff
@@ -0,0 +1,578 @@
+engine will inspect the _engine version_ declared by your plugin and correctly
+prevent your plugin from being used. When those users upgrade their napari
+installation, they'll be able to take advantage of the new functionality you
+added.
+
+## Configuring a python package to use a plugin manifest
+
+### 1. Add package metadata for locating the manifest
+
+The manifest file should be specified in the plugin's `setup.cfg` or
+`setup.py` file using the _[entry point group][epg]_: `napari.manifest`. For
+example, this would be the section for a plugin `npe2-tester` with
+`napari.yaml` as the manifest file:
+
+```cfg
+[options.entry_points]
+napari.manifest =
+    npe2-tester = npe2_tester:napari.yaml
+```
+
+The manifest file is specified relative to the submodule root path.
+So for the example it will be loaded from: `<path/to/npe2-tester>/napari.yaml`.
+
+### 2. Include the manifest in the package distribution
+
+The manifest file needs to be included as _[package data][pd]_ in
+distributable forms for the package. For example:
+
+```toml
+[metadata]
+...
+include_package_data=True
+
+[options.package_data]
+npe2_tester =
+    napari.yaml
+```
+
+## Manifest schema
+
+The plugin manifest file is a [json][], [yaml][], or [toml][] conforming to
+a schema. The `npe2` package includes a command-line tool that can be used
+to validate a schema:
+
+```
+pip install npe2
+npe2 validate my_plugin_manifest.yml
+```
+
+The manifest file's fields are described in detail below. The manifest file's
+overall structure can be viewed as a hierarchy:
+
+```
+<top-level properties>
+contributions:
+  commands:     <list of commands>
+  readers:      <list of readers>
+  writers:      <list of writers>
+  sample_data:  <list of sample data providers>
+  widgets:      <list of widget providers>
+  themes:       <list of themes>
+```
+
+Readers, writers, sample data providers and widget providers refer to callable
+python functions that a plugin defines. Each callable is identified with an
+entry in the list of `commands` via a unique id. For more see {ref}`Commands` below.
+
+```{note}
+Python package metadata (`setup.py` or `setup.cfg`) may be used to populate
+selected properties of the [PluginManifest][].
+```
+
+(top-level-props)=
+
+## Top-level properties
+
+### Required
+
+- **name** The name of the plugin. Example: `napari-svg`. Should be a
+  [PEP-8][]-compatible package name. If missing, this is populated from the
+  python package [name][setup-name].
+
+### Optional
+
+- **engine**: A SemVer compatible version string matching the versions of the
+  plugin engine that the extension is compatible with. Example: "0.1.0".
+- **display_name**: User-facing text to display as the name of this plugin.
+  Example: `napari SVG`. Must be 3-40 characters long, containing
+  printable word characters, and must not begin or end with an underscore,
+  white space, or non-word character.
+- **entry_point**: The module containing the `activate()` function. Example:
```

Comment:
I feel like this is kinda confusing in a world where `activate()` functions are currently not really used? Can we relate this to either the entry point or the manifest or something?

Reply:
@nclack and I had discussed changing this to a python name pointing to the activate function itself..: which is clearer I think

Reply:
I added an admonition describing activation/deactivation.

@tlambert03 I'm ok with entry_point.  activate and deactivate might be clearer here, but I feel like sometime fairly soon we'll take a serious look at dynamic plugin generation and redesign this part anyway.